### PR TITLE
Reader: break code into separate section chunks

### PR DESCRIPTION
**Pull request by mtias**
_In Progress_

Updates Reader routing and controller to separate sections into separate code bundles improving the loading performance.

### CODE REVIEW

**Comment by mtias** [comment]
cc @bluefuton @blowery for review. If we can get this as an initial pass before we start getting conflicts, that's be great. Or if you want to fix some of the duplicated code here, feel free to push to it.

**Comment by bluefuton** [comment]
Nice one @mtias!

It looks like feed and site streams are broken at the moment, e.g.

http://calypso.localhost:3000/read/blogs/82798297
http://calypso.localhost:3000/read/feeds/40474296

screen shot 2016-04-29 at 12 17 47

**Comment by bluefuton** [comment]
Thanks @mtias, feed and blog streams work well now.

Looks like we have a different issue on tag streams though (e.g. http://calypso.localhost:3000/tag/banana):

screen shot 2016-04-29 at 17 55 53

**Comment by blowery** [comment]
I'd like to avoid all the duped code in there. `ensureStoreLoading` is defined in each controller and should really only be in one place.

**Comment by mtias** [comment]
@blowery sure, I didn't know where you wanted to keep these utilities as the controller files don't seem great candidates.

**Comment by blowery** [comment]
> @blowery sure, I didn't know where you wanted to keep these utilities as the controller files don't seem great candidates.

controller is fine for now. we'll switch to query components eventually.

**Comment by mtias** [comment]
If we require `ensureStoreLoading` from the main reader controller we lose the benefit of separating controllers because we'll be loading the whole file for every section.

File: client/reader/discover/controller.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import ReactDom from 'react-dom';
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'lib/analytics';
+import config from 'config';
+import route from 'lib/route';
+import trackScrollPage from 'lib/track-scroll-page';
```
Comment (blowery) [inline client/reader/discover/controller.js]: this is different from the reader's trackScrollPage. It doesn't bump the reader view mc stat.

**Comment by blowery** [comment]
Broke things out into controller-helper and updated things to use them. Really simplified imports nicely, and the debug main reader bundle is now down to 478kb. Nice. Thanks for spearheading this @mtias!

**Comment by mtias** [comment]
Great, thanks. The tags issue is fixed, so feel free to get this merge at your discretion.

**Pull request merged**